rust/pact_verifier/src/pact_broker.rs

```diff
@@ -31,24 +25,23 @@ fn as_string(json: &serde_json::Value) -> String {
     }
 }
 
-fn content_type<T>(response: &Response<T>) -> String {
+fn content_type(response: &reqwest::async::Response) -> String {
     match response.headers().get("content-type") {
         Some(value) => value.to_str().unwrap_or("text/plain").into(),
         None => s!("text/plain")
     }
 }
 
-fn json_content_type<T>(response: &Response<T>) -> bool {
-    match response.headers().get("content-type") {
-        Some(value) => {
-            match value.to_str() {
-                Err(_) => false,
-                Ok("application/json") => true,
-                Ok("application/hal+json") => true,
+fn json_content_type(response: &reqwest::async::Response) -> bool {
+    match content_type(response).parse::<mime::Mime>() {
+        Ok(mime) => {
+            match (mime.type_().as_str(), mime.subtype().as_str(), mime.suffix()) {
+                ("application", "json", None) => true,
+                ("application", "hal", Some(mime::JSON)) => true,
```

Comment:
There's a bug in `mime` that you cannot create a user-defined `mime::Name` (the return type of `type_`, `subtype` and `suffix`). You can only covert a name to a `&str` and not the other way around: https://github.com/hyperium/mime/issues/77. The bug has been fixed, but the `0.4` version is still unreleased.